src/Val.h

```diff
@@ -332,6 +332,9 @@ class ValManager
 
 	inline const StringValPtr& EmptyString() const { return empty_string; }
 
+	// Get a ListVal instance with the given type tag.
```

Comment:
I'm not a huge fan of adding a method for this. Can we not add a new constructor to ListVal that just takes a TypeTag, add a `friend` to ListManager, and have the constructor do a lookup itself? You'd have to remove the default argument from the version that takes a `TypeList` but that would at least give you something explicit to deprecate.

Reply:
Thanks. Let me give it a spin. If that avoids the shotgun updates everywhere :+1:

Reply:
Yeah that's kinda what I was hoping too. This is a really large changeset for something that seems like it should be fairly trivial.